azkaban-common/src/main/java/azkaban/executor/container/ContainerizedExecutionManager.java

```diff
@@ -0,0 +1,383 @@
+      this.executionsBatchProcessingEnabled = azkProps
+          .getBoolean(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_ENABLED,
+              false);
+      this.executionsBatchSize =
+          azkProps
+              .getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_SIZE,
+                  10);
+
+      this.executorService = Executors.newFixedThreadPool(azkProps.getInt(
+          ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_PROCESSING_THREAD_SIZE,
+          10));
+      this.setName("Containerized-QueueProcessor-Thread");
+    }
+
+    @Override
+    public void run() {
+      // Loops till QueueProcessorThread is shutdown
+      while (!this.shutdown) {
+        synchronized (this) {
+          try {
+            // start processing queue if active, otherwise wait for sometime
+            if (this.isActive) {
+              processQueuedFlows();
+            }
+            wait(queueProcessorWaitInMS);
+          } catch (final Exception e) {
+            ContainerizedExecutionManager.logger.error(
+                "QueueProcessorThread Interrupted. Probably to shut down.", e);
+          }
+        }
+      }
+    }
+
+    /* Method responsible for processing the non-dispatched flows */
+    private void processQueuedFlows() throws ExecutorManagerException {
+      Set<Integer> executionIds =
+          executorLoader.selectAndUpdateExecutionWithLocking(this.executionsBatchProcessingEnabled, this.executionsBatchSize,
+              Status.DISPATCHING);
+
+      for (int executionId : executionIds) {
+        Runnable worker = new ExecutionDispatcher(executionId);
+        executorService.execute(worker);
+      }
+    }
+
+    public boolean isActive() {
+      return this.isActive;
+    }
+
+    public void setActive(final boolean isActive) {
+      this.isActive = isActive;
+      ContainerizedExecutionManager.logger
+          .info("QueueProcessorThread active turned " + this.isActive);
+    }
+
+    public void shutdown() {
+      this.shutdown = true;
+      this.executorService.shutdown();
+      this.interrupt();
+    }
+  }
+
+  /**
+   * This class is a worker class to dispatch execution in container.
+   */
+  private class ExecutionDispatcher implements Runnable {
+
+    private final int executionId;
+
+    ExecutionDispatcher(final int executionId) {
+      this.executionId = executionId;
+    }
+
+    @Override
+    public void run() {
+      try {
+        ContainerizedExecutionManager.this.containerizedImpl.createContainer(executionId);
+      } catch (ExecutorManagerException e) {
+        logger.info("Unable to dispatch container in Kubernetes for : {}", executionId);
+        final ExecutableFlow dsFlow;
+        try {
+          dsFlow =
+              ContainerizedExecutionManager.this.executorLoader.fetchExecutableFlow(executionId);
+          dsFlow.setStatus(Status.READY);
```

Comment:
Why are we setting the flow as Ready here? The flow is already in Dispatching state at this point!

Reply:
If the dispatch fails then we need to mark it as READY so it will be back in a queue. It will be picked up again for dispatch.